components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/bean/RuleTemplate.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.business.rules.core.bean;
+
+
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * Represents a Rule Template, which consists of one or more Templates
+ */
+public class RuleTemplate {
+    private String uuid;
+    private String name;
+    private String type; // "input" "output" or "template"
```

Comment:
cant we make this enum?

Reply:
We only use this class to map the ruleTemplate  json, that is reading from the business rule json file. We never manually assign any value to 'type'. So do we need a enum for this?